pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -1033,7 +1034,7 @@ func (ssc *defaultStatefulSetControl) processReplica(
 	// (e.g. terminated on node reboot) is determined by the exit code of the
 	// container, not by the reason for pod termination. We should restart the pod
 	// regardless of the exit code.
-	if isFailed(replicas[i]) || isSucceeded(replicas[i]) {
+	if isFailed(replicas[i]) || isSucceeded(replicas[i]) || specifieddelete.IsSpecifiedDelete(replicas[i]) {
```

Comment:
You can't add the condition here, because`specified-delete` pods also should be protected by `maxUnavailable`.

Reply:
You're right